src/lmdb.mli

```diff
@@ -43,25 +43,19 @@ module Mdb :module type of Lmdb_bindings
     cursors. The following values are used to request read-only or read-write
     permissions on environments, transactions and cursors.
 *)
-
-type 'a perm constraint 'a = [< `Read | `Write ]
-
-(** Request read-only access. *)
-val ro : [ `Read ] perm
-
-(** Request read-write access. *)
-val rw : [ `Read | `Write ] perm
-
+type 'a perm =
+  | Ro : [ `Read ] perm
+  | Rw : [ `Read | `Write ] perm
```

Comment:
I kept this type opaque because
1. I had to struggle a lot to get the pattern matchings an the GADT to type as I wanted.
2. We might indeed want a different internal representation
(e.g. ``type 'a perm = [> `Read | `Write | `Dup] array)``
3. No benefit from exposing this type.

Reply:
The reason I exposed it is to 1) avoid having *any* value in `Lmdb` and 2) leverage type based disambiguation.

This way, `Lmdb.Env.create ~perm:Ro ...` works, via disambiguation.